### PR TITLE
Refactor micropkg and remove pipeline pull and pipeline package

### PR DESCRIPTION
**Pull request by SajidAlamQB**
## Description

Following https://github.com/kedro-org/kedro/pull/1224 we need to refactor some variable names and functions so that they don't talk about pipelines but more generally about micropackages. We also need to remove `pipeline package` and `pipeline pull` and their tests as part of this issue.

## Development notes
- Removed `pipeline package` and `pipeline pull` commands and their tests
- Renamed mentions of `pipeline` to `micropkg` in the docs/notes/code

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1328"></a>

### CODE REVIEW

File: kedro/framework/cli/micropkg.py [hunk 1]
```diff
@@ -267,7 +280,219 @@ def package_pipeline(
...
+        |__ <path_to_pipeline>
+            |__ __init__.py
+            |__ <pipeline_name>
+                |__ __init__.py
+    |__ tests
+        |__ __init__.py
+        |__ <path_to_pipeline>
+            |__ __init__.py
+            |__ <pipeline_name>
+                |__ __init__.py
```
Comment (lorenabalan) [inline kedro/framework/cli/micropkg.py:368]: Might be nice to update these docstrings too, to refer to `project_package` and `micro-package` etc.

**Review by lorenabalan — APPROVED** [review]
I'm surprised we didn't have to add or change some micropkg unit tests, but it looks like we just moved some functions around instead of importing them, right? So the functionality was already correct on `develop`?

**Comment by SajidAlamQB** [comment]
> I'm surprised we didn't have to add or change some micropkg unit tests, but it looks like we just moved some functions around instead of importing them, right? So the functionality was already correct on `develop`?

Right, just moved all the functions not being used by `pipeline create/delete` into `micropkg.py` and imported the remaining functions as needed.

**Review by AntonyMilneQB — APPROVED** [review]
Looks good, but I think there's a few places where we could probably replace `pipeline` with `micopkg` still left, e.g. throughout `_refactor_code_for_unpacking`, `_get_default_version`, `_package_micropkg`, `_refactor_code_for_package`, `_generate_sdist_file`. Basically if I just ctrl+F for "pipeline" in micropkg.py I don't there's many places it should still appear.

At some point in the future I think we should rework the micropackaging documentation also, but that's for another time.

**Comment by MerelTheisenQB** [comment]
> Looks good, but I think there's a few places where we could probably replace `pipeline` with `package`/`micropkg` still left, e.g. throughout `_refactor_code_for_unpacking`, `_get_default_version`, `_package_micropkg`, `_refactor_code_for_package`, `_generate_sdist_file`. Basically if I just ctrl+F for "pipeline" in micropkg.py I don't there's many places it should still appear.

While I was doing the changes I thought some of it was still `pipeline` specific, but on second thought you're right and I'll update it!

**Pull request merged**